subprojects/packagefiles/theora/tests/meson.build

```diff
@@ -0,0 +1,109 @@
+theora_libs = [libtheora]
+theoradec_libs = [libtheoradec]
+theoraenc_libs = [libtheoraenc, libtheoradec]
+
+tests = {}
+
+# dummy call tests for the current api
+noop = executable(
+  'noop',
+  files('noop.c'),
+  link_with: theoraenc_libs,
+  dependencies: [m_dep, ogg_dep, config_dep],
+  include_directories: incdir,
+)
+
+tests += {'noop': noop}
+
+# dummy call tests for the pre-1.0 legacy api with current link line
+noop_theoraenc = executable(
+  'noop_theoraenc',
+  files('noop_theora.c'),
+  link_with: theoraenc_libs,
+  dependencies: [m_dep, ogg_dep, config_dep],
+  include_directories: incdir,
+)
+
+tests += {'noop_theoraenc': noop_theoraenc}
+
+# dummy call tests for the pre-1.0 legacy api with legacy link line
+noop_theora = executable(
+  'noop_theora',
+  files('noop_theora.c'),
+  link_with: theora_libs,
+  dependencies: [m_dep, ogg_dep, config_dep],
+  include_directories: incdir,
+)
+
+tests += {'noop_theora': noop_theora}
+
+# comment utilities for the current api
+comment = executable(
+  'comment',
+  files('comment.c'),
+  link_with: theoradec_libs,
+  dependencies: [m_dep, ogg_dep, config_dep],
+  include_directories: incdir,
+)
+
+tests += {'comment': comment}
+
+# comment utilities for the legacy api and current lib
+comment_theoradec = executable(
+  'comment_theoradec',
+  files('comment.c'),
+  link_with: theoradec_libs,
+  dependencies: [m_dep, ogg_dep, config_dep],
+  include_directories: incdir,
+)
+
+tests += {'comment_theoradec': comment_theoradec}
+
+# comment utilities for the legacy api and legacy lib
+comment_theora = executable(
+  'comment_theora',
+  files('comment_theora.c'),
+  link_with: theora_libs,
+  dependencies: [m_dep, ogg_dep, config_dep],
+  include_directories: incdir,
+)
+
+tests += {'comment_theora': comment_theora}
+
+granulepos = executable(
+  'granulepos',
+  files('granulepos.c'),
+  link_with: theoraenc_libs,
+  dependencies: [m_dep, ogg_dep, config_dep],
+  include_directories: incdir,
+)
+
+tests += {'granulepos': granulepos}
+
+granulepos_theoraenc = executable(
+  'granulepos_theoraenc',
+  files('granulepos_theora.c'),
+  link_with: theoraenc_libs,
+  dependencies: [m_dep, ogg_dep, config_dep],
+  include_directories: incdir,
+)
+
+tests += {'granulepos_theoraenc': granulepos_theoraenc}
+
+granulepos_theora = executable(
+  'granulepos_theora',
+  files('granulepos_theora.c'),
+  link_with: theora_libs,
+  dependencies: [m_dep, ogg_dep, config_dep],
+  include_directories: incdir,
+)
+
+tests += {'granulepos_theora': granulepos_theora}
+
+foreach name, exe : tests
+  if conf.get('HAVE_VALGRIND', false)
+    test(name, valgrind_exe, args: test_env + exe)
+  else
+    test(name, exe)
+  endif
+endforeach
```

Comment:
I wonder if maybe this wouldn't be better off as:

```meson
if valgrind_exe.found()
    add_test_setup('valgrind', exe_wrapper: [valgrind] + test_env)
endif
```

You could then run the entire test suite with `meson test --setup=valgrind` or even pass `is_default: true` to the test setup definition.

...


While I'm on that topic, why is HAVE_VALGRIND a config.h header define? It's certainly not used in the codebase according to my code search. Maybe this doesn't even need an option in meson_options.txt -- simply look for it if there, and if so, define a test setup but not by default.

Reply:
> While I'm on that topic, why is HAVE_VALGRIND a config.h header define?

It's a carry-over from my FFmpeg wrap inspiration, tossing everything into the configure object. If Meson has test setups (TIL!) then I should be able to simplify all this.